diff --git a/mirakc-core/src/epg/models.rs b/mirakc-core/src/epg/models.rs
index 984a1544..9a96c575 100644
--- a/mirakc-core/src/epg/models.rs
+++ b/mirakc-core/src/epg/models.rs
@@ -26,7 +26,7 @@ pub struct EitSection {
 
 impl EitSection {
     pub fn is_valid(&self) -> bool {
-        matches!(self.table_id, 0x50 | 0x51 | 0x58 | 0x59)
+        self.table_id >= 0x50 && self.table_id <= 0x5F
     }
 
     pub fn is_basic(&self) -> bool {
